File: control-flow/Q129/main.go
package main

import "fmt"

func main() {
	t := true
	f := false
	fmt.Println("Logical AND &&")
	fmt.Printf("%t && %t: %d\n", t, t, boolToInt(t && t))
	fmt.Printf("%t && %t: %d\n", t, f, boolToInt(t && f))
	fmt.Printf("%t && %t: %d\n", f, t, boolToInt(f && t))
	fmt.Printf("%t && %t: %d\n", f, f, boolToInt(f && f))
	fmt.Println("Logical OR ||")
	fmt.Printf("%t || %t: %d\n", t, t, boolToInt(t || t))
	fmt.Printf("%t || %t: %d\n", t, f, boolToInt(t || f))
	fmt.Printf("%t || %t: %d\n", f, t, boolToInt(f || t))
	fmt.Printf("%t || %t: %d\n", f, f, boolToInt(f || f))
	fmt.Println("Logical NOT !")
	fmt.Printf("!%t: %d\n", t, boolToInt(t))
	fmt.Printf("!%t: %d\n", f, boolToInt(f))
}

func boolToInt(v bool) int {
	if v {
		return 1
	}
	return 0
}

File: control-flow/Q129/main.rs
use std::io;

fn bool_to_int(v: bool) -> i32 {
    if v { 1 } else { 0 }
}

fn main() {
    let _ = io::stdin();

    let t = true;
    let f = false;

    println!("Logical AND &&");
    println!("{} && {}: {}", t, t, bool_to_int(t && t));
    println!("{} && {}: {}", t, f, bool_to_int(t && f));
    println!("{} && {}: {}", f, t, bool_to_int(f && t));
    println!("{} && {}: {}", f, f, bool_to_int(f && f));

    println!("Logical OR ||");
    println!("{} || {}: {}", t, t, bool_to_int(t || t));
    println!("{} || {}: {}", t, f, bool_to_int(t || f));
    println!("{} || {}: {}", f, t, bool_to_int(f || t));
    println!("{} || {}: {}", f, f, bool_to_int(f || f));

    println!("Logical NOT !");
    println!("!{}: {}", t, bool_to_int(!t));
    println!("!{}: {}", f, bool_to_int(!f));
}